plugin/evm/orderbook/memory_database.go

```diff
@@ -615,8 +616,12 @@ func (db *InMemoryDatabase) UpdateFilledBaseAssetQuantity(quantity *big.Int, ord
 		order.LifecycleList = order.LifecycleList[:len(order.LifecycleList)-1]
 	}
 
+	// only update margin if the order is not reduce-only
 	if order.OrderType == Signed && !order.ReduceOnly {
-		// only update margin if the order is not reduce-only
+		// for short orders, quantity is negative when filled and positive when event was removed
+		if order.PositionType == SHORT {
+			quantity = hu.Neg(quantity)
+		}
```

Comment:
this is not correct. `quantity` is always +ve when order is matched and -ve when the event reorgs.

Reply:
right, it was incorrect. Fixed it now